lib/_stream_readable.js

```diff
@@ -758,8 +758,10 @@ Readable.prototype.unpipe = function(dest) {
     state.pipesCount = 0;
     state.flowing = false;
 
-    for (var i = 0; i < len; i++)
+    for (var i = 0; i < len; i++) {
+      unpipeInfo = { hasUnpiped: false };
```

Comment:
For other reviewers,`dest` sets unpipeInfo.hasUnpiped to `false` when `'unpipe'` is emitted.

Reply:
On second thought, doesn't this make the `hasUnpiped` guard useless? I mean, no matter what, `dest` will always find `hasUnpiped` `false`.

Reply:
I found [the PR](https://github.com/nodejs/node/pull/12746) which adds the `hasUnpiped`. you can see it.

Reply:
@lpinca `hasUnpiped` solved the situation that a readable stram piped the same writable stream more than one time. This change has no influence to it.

Reply:
@lpinca `hasUnpiped` solved the situation that a readable stram piped the same writable stream more than one time. This change has no influence to it.

Reply:
Ok yes, it seems correct as this is for the "remove all" case.